### ISSUE TITLE
Worlds are broken

### ISSUE THREAD

**Issue by AzureBerylBlue** [mmatyas/supermariowar#81]
Compiled the latest version, and worlds are broken. All levels take place in 0smw. You must have broke this at some point.

**Comment by mmatyas** [comment]
Indeed, it seems like there are some issues with the tour file reading.

**Comment by AzureBerylBlue** [comment]
Fixed it by reverting this commit: 76a72feb9229a4d811b495c5a91dd85302de36e9

**Comment by mmatyas** [comment]
Ah I see. I wonder if it could be fixed without losing the functionality that this commit provides?

**Comment by AzureBerylBlue** [comment]
I think your commit is expecting the names of the levels in the world and tour .txt files to have the creator and .map file extension on it. You need the opposite of stripCreatorAndDotMap() when reading the level names from world and tour files somehow.

**Comment by AzureBerylBlue** [comment]
If you go to single game, change the map to something else, back out, and play a tour/world, all the levels will be set to the last level you looked at, not 0smw. It's like some variable somewhere is not being set.
